src/game/tests/test_grid.py

```diff
@@ -32,6 +33,46 @@ def test_grid_to_string_should_create_the_reference_string(self):
 """  # noqa
         self.assertEqual(grid_display, reference_display)
 
+    def test_load_state_with_invalid_string_create_empty_grid(self):
+        arg = []
+        arg.append("""--state={"grid" : {"A2": 10,"C1":3,"D1":12},"turn" :{"player" : 1,"selected" : 7}""")
+        parameter, error_message = get_state_parameter(arg)
+        game_state = GameState(parameter)
+        i = 0
+        nb_dots = 0
+        while i < 4:
+            nb_dots += game_state.grid[i].count('.')
+            i += 1
+        self.assertEqual(nb_dots, GRID_SIZE*GRID_SIZE)
+
+    def test_load_state_with_valid_string_place_correct_piece_on_grid(self):
+        arg = []
+        arg.append("""--state={"grid" : {"A2": 10,"C1":3,"D1":12},"turn" :{"player" : 1,"selected" : 7}}""")
+        parameter, error_message = get_state_parameter(arg)
+        game_state = GameState(parameter)
+        self.assertEqual(game_state.grid[1][0], 10)
+
+    def test_load_state_with_valid_string_remove_correct_piece_from_remaining_list(self):
+        arg = []
+        arg.append("""--state={"grid" : {"A2": 10,"C1":3,"D1":12},"turn" :{"player" : 1,"selected" : 7}}""")
+        parameter, error_message = get_state_parameter(arg)
+        game_state = GameState(parameter)
+        self.assertEqual(game_state.remaining_pieces.count(12), 0)
+
+    def test_load_state_with_valid_string_select_correct_player(self):
+        arg = []
+        arg.append("""--state={"grid" : {"A2": 10,"C1":3,"D1":12},"turn" :{"player" : 2,"selected" : 7}}""")
+        parameter, error_message = get_state_parameter(arg)
+        game_state = GameState(parameter)
+        self.assertEqual(game_state.game_turn.player_one_active, False)
+
+    def test_load_state_with_valid_string_select_correct_piece_to_play(self):
+        arg = []
```

Comment:
Not mandatory for this review but ... 

```py
arg = []
arg.append("""--state={"grid" : {"A2": 10,"C1":3,"D1":12},"turn" :{"player" : 2,"selected" : 7}}""")
```
===
```py
arg = ["""--state={"grid" : {"A2": 10,"C1":3,"D1":12},"turn" :{"player" : 2,"selected" : 7}}"""]
```

Reply:
thanks, I change it for remember this syntax possibility later